scripts/commands/updateApiDocs.ts

```diff
@@ -318,22 +318,25 @@ async function convertHtmlToMarkdown(
 
   for (const result of results) {
     let path = urlToPath(result.url);
-    if (pkg.hasSeparateReleaseNotes && path.endsWith("release-notes.md")) {
-      // Historical versions use the same release notes files as the current API
-      if (pkg.historical) {
-        continue;
-      }
-
-      // Convert the relative links to absolute links
-      result.markdown = transformLinks(result.markdown, (link, _) =>
-        link.startsWith("http") || link.startsWith("#") || link.startsWith("/")
-          ? link
-          : `/api/${pkg.name}/${link}`,
-      );
 
-      path = currentReleaseNotesPath(pkg);
+    if (!pkg.hasSeparateReleaseNotes || !path.endsWith("release-notes.md")) {
+      await writeFile(path, result.markdown);
+      continue;
+    }
+
+    // Historical versions use the same release notes files as the current API
+    if (pkg.historical) {
+      continue;
     }
```

Comment:
Oh hm, I'm no longer convinced this is true. Imagine that current is 1.0.0, which we released last week. We just released 0.46.2 today, and now we're generating 0.46.2 with `--historical`. The release notes from last week for 1.0.0 will not yet include 0.46.2 because it didn't exist at the time.

I think that we do need to consider historical release notes. We only need to be careful to not delete the notes for future versions; when generating `-v 0.46.2 --historical`, we shouldn't touch `release-notes/1.0.md`.

--

I'm fine if this is fixed in a followup. This PR already has some complexity and it might be easier to address this edge case in a dedicated change.